src/components/decimal/decimal.component.tsx

```diff
@@ -179,6 +158,30 @@ export const Decimal = React.forwardRef(
       [getSeparator, isNaN, l, locale, precision]
     );
 
+    const emptyValueWithPrecision =
+      precision === 0 ? "0" : formatValue(`0.${"0".repeat(precision)}`);
+
+    const emptyValue = allowEmptyValue ? "" : emptyValueWithPrecision;
+
+    const getSafeValueProp = useCallback(
+      (initialValue) => {
+        // We're intentionally preventing the use of number values to help prevent any unintentional rounding issues
+        invariant(
+          typeof initialValue === "string",
+          "Decimal `value` prop must be a string"
+        );
+
+        if (initialValue && !allowEmptyValue) {
+          invariant(
+            initialValue !== "",
+            "Decimal `value` must not be an empty string. Please use `allowEmptyValue` or `0.00`"
```

Comment:
**question**: not exactly a suggestion as I'm not sure what would be better, but I'm not sure this message is accurate any more - the part about `0.00` I mean.

I think the message is confusing anyway, unless I'm missing something (pretty likely!) I think the best message is just something like "Please either set `allowEmptyValue` or specify a non-empty `initialValue`.".

Reply:
I think you make a good point. This invariant message comes from when Decimal was first written about 4 years ago now. [See it here](https://github.com/Sage/carbon/blame/5bf51f7a32490113141308b2e02028402fd3fbcc/src/__experimental__/components/decimal/decimal.component.js#L193). We can probably change it to your suggestion as that makes more sense considering the change on this PR.